pkg/util/kubernetes/hostinfo/tags.go

```diff
@@ -18,44 +18,61 @@ import (
 )
 
 // GetTags gets the tags from the kubernetes apiserver and the kubelet
-func GetTags(ctx context.Context) (tags []string, err error) {
-	tags = appendNodeInfoTags(ctx, tags)
+func GetTags(ctx context.Context) ([]string, error) {
+	var tags []string
+	tags, err := appendNodeInfoTags(ctx, tags)
+	if err != nil {
+		return nil, err
+	}
 
 	annotationsToTags := getAnnotationsToTags()
-	if len(annotationsToTags) > 0 {
-		nodeAnnotations, e := GetNodeAnnotations(ctx)
-		if e != nil {
-			err = e
-		} else {
-			tags = append(tags, extractTags(nodeAnnotations, annotationsToTags)...)
-		}
+	if len(annotationsToTags) == 0 {
+		return tags, nil
 	}
 
-	return
+	nodeAnnotations, err := GetNodeAnnotations(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tags = append(tags, extractTags(nodeAnnotations, annotationsToTags)...)
+
+	return tags, nil
 }
 
-func appendNodeInfoTags(ctx context.Context, tags []string) []string {
+func appendNodeInfoTags(ctx context.Context, tags []string) ([]string, error) {
 	nodeInfo, err := NewNodeInfo()
 	if err != nil {
 		log.Debugf("Unable to auto discover node info tags: %s", err)
-		return tags
+		return nil, err
 	}
```

Comment:
The behavior of this function in case of error is changing.

Before, in case of error, the function was returning the unmodified `tags` input parameter.
Now, it is returning `nil`.

The caller of this function is most probably expecting `tags` to be augmented, but never stripped.

In this precise case, it doesn’t make a big difference since the only caller is `GetTags` which is always passing an empty slice as `tags` parameter.
So, maybe we can completely remove this input parameter.

Reply:
Nice catch ! Done in the last commit 👍